controller/persistence/api_session_store.go

```diff
@@ -106,6 +109,33 @@ type apiSessionStoreImpl struct {
 	symbolIdentity boltz.EntitySymbol
 }
 
+func (store *apiSessionStoreImpl) Create(ctx boltz.MutateContext, entity boltz.Entity) error {
+	err := store.baseStore.Create(ctx, entity)
+
+	if err == nil {
+		if apiSession, ok := entity.(*ApiSession); ok && apiSession != nil {
+			if apiSession.MfaRequired == false || apiSession.MfaComplete == true {
+				store.Emit(EventFullyAuthenticated, apiSession)
+			}
+		}
+	}
+
+	return err
+}
+func (store *apiSessionStoreImpl) Update(ctx boltz.MutateContext, entity boltz.Entity, checker boltz.FieldChecker) error {
+	err := store.baseStore.Update(ctx, entity, checker)
+
+	if err == nil {
+		if apiSession, ok := entity.(*ApiSession); ok && apiSession != nil {
+			if (checker == nil || checker.IsUpdated(FieldApiSessionMfaComplete)) && apiSession.MfaComplete == true {
```

Comment:
I was not expecting if `checker == nil` we allow updating all fields, but that is what the lower `ProceedWithSet` function does.